dangerzone/isolation_provider/qubes.py

```diff
@@ -0,0 +1,173 @@
+import asyncio
+import glob
+import inspect
+import io
+import logging
+import os
+import shutil
+import subprocess
+import sys
+import tempfile
+import time
+import zipfile
+from pathlib import Path
+from typing import Callable, Optional
+
+from ..document import Document
+from ..util import get_resource_path
+from .base import IsolationProvider
+
+log = logging.getLogger(__name__)
+
+from ..conversion.pixels_to_pdf import PixelsToPDF
+from ..util import get_subprocess_startupinfo, get_tmp_dir
+
+CONVERTED_FILE_PATH = (
+    "/tmp/safe-output-compressed.pdf"  # FIXME won't work for parallel conversions
+)
+
+
+class Qubes(IsolationProvider):
+    """Uses a disposable qube for performing the conversion"""
+
+    def install(self) -> bool:
+        pass
+
+    def _convert(
+        self,
+        document: Document,
+        ocr_lang: Optional[str],
+        stdout_callback: Optional[Callable] = None,
+    ):
+        success = False
+
+        # FIXME won't work on windows, nor with multi-conversion
+        out_dir = Path("/tmp/dangerzone")
+        if out_dir.exists():
+            shutil.rmtree(out_dir)
+        out_dir.mkdir()
+
+        # Reset hard-coded state
+        if os.path.exists(CONVERTED_FILE_PATH):
+            os.remove(CONVERTED_FILE_PATH)
+
+        percentage = 0.0
+
+        with open(document.input_filename, "rb") as f:
+            # TODO handle lack of memory to start qube
+            if getattr(sys, "dangerzone_dev", False):
+                # Use dz.ConvertDev RPC call instead, if we are in development mode.
```

Comment:
> @deeplow: This whole section feels a bit like ti could have its own function (update_code_in_dvm() or something)
> @apyrgio: Under a qubes module, most probably. You're right, I'd like something similar.


(from [here](https://github.com/freedomofpress/dangerzone/commit/ade3e2280a4303569626542302c3a770bf6726dd#r116691345))

Reply:
Let's leave this as-is and tackle it later via https://github.com/freedomofpress/dangerzone/issues/442

Reply:
> What would work is giving this function the write end of the pipe, so that it can write to it.

OK. Feel free to adapt it or write it from scratch.

Reply:
We now have code for this in 5456f8b2. It should tackle the buffer size limitation.